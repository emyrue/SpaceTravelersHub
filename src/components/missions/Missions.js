import { useEffect } from 'react';
import Table from 'react-bootstrap/Table';
import { useDispatch, useSelector } from 'react-redux';
import { getMissions } from '../../redux/missions/missions';
import MissionData from './MissionData';

const Missions = () => {
  const missions = useSelector((state) => state.missions);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getMissions());
  }, []);

  return (
    <div>
      <h1>Missions</h1>

      <Table striped bordered>
        <thead>
          <tr>
            <th>Mission</th>
            <th>Description</th>
            <th>Status</th>
            <th>&nbsp;</th>
          </tr>
        </thead>

        <tbody>
          {missions.map((mission) => (
            <MissionData key={mission.id} mission={mission} />
          ))}
        </tbody>

      </Table>
    </div>
  );
};

export default Missions;
