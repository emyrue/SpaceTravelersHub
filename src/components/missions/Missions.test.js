import renderer from 'react-test-renderer';
import Missions from './Missions';
import { Provider } from 'react-redux';
import store from '../../redux/configureStore';

describe(' Missions Component Rendering ', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Missions />
        </Provider>)
    expect(tree).toMatchSnapshot();
  });
})