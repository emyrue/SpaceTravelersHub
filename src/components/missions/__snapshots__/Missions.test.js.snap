// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Missions Component Rendering  renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="py-5"
  >
    <table
      className="table table-striped table-bordered"
    >
      <thead>
        <tr>
          <th>
            Mission
          </th>
          <th>
            Description
          </th>
          <th>
            Status
          </th>
          <th>
            Â 
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;
